app/src/app/bookmarks/BookmarkList.tsx

```diff
@@ -1,39 +1,40 @@
 import { useState, useEffect } from "react";
-import { Pagination, Table } from "react-bootstrap";
+import { Table } from "react-bootstrap";
 import BookmarkBtn, { getBookmarks } from "./BookmarkBtn";
 import { AiOutlineLink } from "react-icons/ai";
 import "./list.css";
+import Pagination from "@components/pagination";
+import { API } from "@/types/ApiRoute";
 
 //
 // This component requires button to redirect to /bookmarks
 // NOTE: See BookmarkListOffCanvas for advanced development
 //
+
 export default function BookmarkList() {
     const [bookmarks, setBookmarks] = useState([]);
     const [page, setPage] = useState(1);
-
-    const next = () => {
-        setPage((page) => ++page);
-    }
-    const prev = () => {
-        setPage((page) => --page);
-    }
+    const [numPages, setNumPages] = useState<number>(1);
 
     useEffect(() => {
-        // We should display error toast if it fails.
-        // page should never be negative,
-        // but if it happens, this will throw
+        fetch(`${API}/pages?optn=bkmk`)
+          .then(response => response.json())
+          .then(data => setNumPages(data.data.total ?? 1))
+          .catch(err => console.log(err))
+    }, [bookmarks])
+
+    useEffect(() => { 
         getBookmarks(page)
             .then((data) => setBookmarks(data.data))
-            .catch((err) => {
-                console.log(err);
-            });
-    }, [page]);
+            .catch((err) => console.log(err))
+    }, [page])
 
     const removeBookmark = (pid: number) => {
-        const newBookmarks = bookmarks.filter((i) => i["pid"] !== pid);
-        setBookmarks(newBookmarks);
-    };
+        bookmarks.filter((i) => i["pid"] !== pid);
```

Comment:
This line is also no longer required. Filter returns a copy of bookmarks that does not match the pid, however the DB already has the update, so the follow lines will update bookmarks anyways